### ISSUE TITLE
Syntax highlighting of code in other languages than Ruby?

### ISSUE THREAD

**Issue by thomthom** [lsegal/yard#1405]
I'm working on adding additional documentation in a `.md` document that ships with the API docs. In the docs I have C/C++ code examples for Ruby C Extensions. However, these code blocks doesn't get any syntax coloring. Is it possible with YARD? Via some addon gem or something?

**Comment by thomthom** [comment]
Also, _where_ does one add a `html_syntax_highlight_LANGNAME` type method?

**Comment by thomthom** [comment]
Thank you for the insights.

And format is only extracted via `!!!LANGNAME` notation, not via Markdown code block notation:

    ```c++
    // ...
    ```

**Comment by lsegal** [comment]
> And format is only extracted via !!!LANGNAME notation, not via Markdown code block notation:

Nope, both work, `!!!LANGNAME` is just a syntax to support indented blocks in RDoc formatting, which doesn't support language annotations (since it has no concept of fencing). YARD will post-process HTML output to detect the language type generated by Markdown and other processors:

https://github.com/lsegal/yard/blob/d61dac0fffb97c1fb40e8170e229d359f1ba31c6/lib/yard/templates/helpers/html_helper.rb#L203-L206

**Comment by lsegal** [comment]
> The HTML that YARD parses is having code blocks that are plain `<pre><code>...</code></pre>`

It would be a requirement for the Markdown processor library you're using to generate language-aware fenced code blocks. YARD doesn't do the generation for markdown, it's up to the library to understand fencing syntax-- since it's not a standard Markdown feature, not all libraries support it. Redcarpet (YARD's default recommended lib) should support this.

**Comment by topherfangio** [comment]
Two questions:

1. I'd like to use https://highlightjs.org/ to get that automatic highlighting. How would you recommend I go about injecting it?
2. Using the `@example` tag, there doesn't seem to be a way to set the language?

```ruby
#
# @example Basic Chat
#    !!!haml
#    = daisy_chat do |chat|
#      - chat.with_message do
#        I can't believe it's not the weekend yet!
```

This renders ok, but doesn't add the `haml` class to the output (it just has `example` and `code` classes).

**Comment by lsegal** [comment]
> [!NOTE]
> This is a batch message, apologies if this feels impersonal.

I am closing out a few stale issues and PRs in the YARD repo and this was flagged for automatic closure because the issue may have been fixed or is likely not planned due to the issue/PR age. If this is still an issue or you believe the issue/PR is still worth a discussion, please feel free to comment or reopen a new issue. Thanks and sorry for not getting around to this the first time!
